exp/controllers/azuremachinepool_controller.go

```diff
@@ -390,6 +393,10 @@ func (ampr *AzureMachinePoolReconciler) reconcileDelete(ctx context.Context, mac
 		if err := amps.Delete(ctx); err != nil {
 			return reconcile.Result{}, errors.Wrapf(err, "error deleting AzureMachinePool %s/%s", clusterScope.Namespace(), machinePoolScope.Name())
 		}
+	} else {
```

Comment:
This part is a bit tricky since we need to keep the AzureMachinePool from being deleted until all its AMPMs are gone even when the resource group is being deleted. This is because otherwise we will be unable to construct the machinepool scope in the AMPM controller.

Reply:
can we not use a ownerReference/finalizer for this?